File: stepik-java-api/src/main/java/org/stepik/api/queries/recommendations/StepikReactionsPostQuery.java
package org.stepik.api.queries.recommendations;

import org.jetbrains.annotations.NotNull;
import org.stepik.api.actions.StepikAbstractAction;
import org.stepik.api.objects.recommendations.ReactionValues;
import org.stepik.api.objects.recommendations.Reactions;
import org.stepik.api.objects.recommendations.ReactionsPost;
import org.stepik.api.queries.StepikAbstractPostQuery;
import org.stepik.api.urls.Urls;

/**
 * @author meanmail
 */
public class StepikReactionsPostQuery extends StepikAbstractPostQuery<Reactions> {
    private final ReactionsPost reactions = new ReactionsPost();

    public StepikReactionsPostQuery(StepikAbstractAction stepikAction) {
        super(stepikAction, Reactions.class);
    }

    @NotNull
    public StepikReactionsPostQuery user(long user) {
        reactions.getRecommendationReaction().setUser(user);
        return this;
    }

    @NotNull
    public StepikReactionsPostQuery lesson(long lesson) {
        reactions.getRecommendationReaction().setLesson(lesson);
        return this;
    }

    @NotNull
    public StepikReactionsPostQuery reaction(@NotNull ReactionValues reaction) {
        reactions.getRecommendationReaction().setReaction(reaction.getValue());
        return this;
    }

    @NotNull
    @Override
    protected String getBody() {
        return getJsonConverter().toJson(reactions);
    }

    @NotNull
    @Override
    protected String getUrl() {
        return Urls.RECOMMENDATION_REACTIONS;
    }
}

File: stepik-java-api/src/main/kotlin/org/stepik/api/queries/recommendations/StepikReactionsPostQuery.kt
package org.stepik.api.queries.recommendations

import org.stepik.api.actions.StepikAbstractAction
import org.stepik.api.objects.recommendations.ReactionValues
import org.stepik.api.objects.recommendations.Reactions
import org.stepik.api.objects.recommendations.ReactionsPost
import org.stepik.api.queries.StepikAbstractPostQuery

class StepikReactionsPostQuery(stepikAction: StepikAbstractAction) :
        StepikAbstractPostQuery<Reactions>(stepikAction, Reactions::class.java) {
    
    private val reactions = ReactionsPost()
    
    fun user(user: Long): StepikReactionsPostQuery {
        reactions.recommendationReaction.user = user
        return this
    }
    
    fun lesson(lesson: Long): StepikReactionsPostQuery {
        reactions.recommendationReaction.lesson = lesson
        return this
    }
    
    fun reaction(reaction: ReactionValues): StepikReactionsPostQuery {
        reactions.recommendationReaction.reaction = reaction.value
        return this
    }
    
    override val body: String
        get () {
            return jsonConverter.toJson(reactions, false)
        }
    
    override val url = "${stepikAction.stepikApiClient.host}/api/recommendation-reactions"
    
}
